docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -0,0 +1,222 @@
+	In terms of when to take a snapshot, a day is given as a general example which should be good for typical use but the specification of how often will be customizable as we know that certain operators have different strategies and use cases for how they’re running their services on our clusters. 
+In terms of how we choose a node as the blessed one, lots of different metrics or heuristics can be used like manually creating a temporary prototype for testing, having the fastest ready time, or anything else which is seen as typically good but a more generalist approach is outlined here since more specific methods may not be as helpful for certain operators. 
+For architectural details of where else the code could exist, the controller section makes the most sense since this proposal will be constantly modifying the state of our objects, but theoretically it could be largely put into hack with shell scripts and then a controller could simply be ordered to trigger that shell script, but this is less maintainable in the long run and not as preferred. 
+
+## Upgrade Strategy
+
+Turning off or on the feature is all that is required to keep previous behavior or make use of the enhancement. No backwards compatibility will be broken, all this feature request will do is add a new controller which can be utilized or not as desired.
+
+## Additional Details
+
+### Test Plan [optional]
+
+There will be e2e tests, at least one of which will be as follows:
+ Have an example node and an example patch, apply the patch to the preexisting node, and then trigger the controller to pretend the interval of time has passed, and then it should attempt to create another node and compare the OS image of the new node and the original node, finding that they are both the same image.
+It should be tested primarily in isolation as other components shouldn’t affect what it tries to do, but it may need to be checked with other components to see what happens if certain race conditions or updates at the same time of AzureMachineTemplate are occurring (in which case a lower priority should likely be assigned to this controller for finishing its task after as ideally those changes are in effect before isolating the node).
+
+### Graduation Criteria [optional]
+
+alpha - The feature is initially constructed and toggleable with a flag
+beta - The feature has e2e tests implemented and is more integrated and seamless with the project
+stable - The feature has been used for a while and is widely acceptable as well as reliable and will now be enabled by default.
+
+### Version Skew Strategy [optional]
+
+The feature itself should not depend significantly on the version of CAPI and will be backwards compatible with old versions of CAPZ since it will be a toggleable feature, so if there is a drift in CAPI and CAPZ versions the functionality should stay the same without breaking anything.
```

Comment:
```suggestion
The feature itself should not depend significantly on the version of CAPI and will be backwards compatible with old versions of CAPZ since it will be a toggleable feature. If there is a drift in CAPI and CAPZ versions, the functionality should stay the same without breaking anything.
```

nit